applications/exporter/application_round_result_exporter.py

```diff
@@ -0,0 +1,194 @@
+                ],
+            )
+            .select_related(
+                "application",
+                "application__organisation",
+                "application__contact_person",
+                "application__application_round",
+            )
+            .prefetch_related(
+                # All reservation unit options ordered by preferred order
+                Prefetch(
+                    "reservation_unit_options",
+                    ReservationUnitOption.objects.select_related(
+                        "reservation_unit",
+                        "reservation_unit__unit",
+                    ).order_by("preferred_order"),
+                ),
+                Prefetch(
+                    "reservation_unit_options__allocated_time_slots",
+                    AllocatedTimeSlot.objects.alias(
+                        day_of_the_week_number=L("day_of_the_week_number"),
+                    ).order_by(
+                        "day_of_the_week_number",
+                        "begin_time",
+                        "end_time",
+                    ),
+                ),
+            )
+            .order_by("application_id", "id")
+            .distinct()  # Avoid duplicate rows due to implicit joins
+        )
+
+    def _get_single_row_data(self, section: ApplicationSection) -> Iterable[ApplicationSectionExportRow]:
+        row = ApplicationSectionExportRow(
+            application_id=str(section.application.id),
+            application_status=section.application_status,  # type: ignore[attr-defined]
+            section_id=str(section.id),
+            section_status=section.status,
+            section_name=section.name,
+            reservations_begin_date=local_date_string(section.reservations_begin_date),
+            reservations_end_date=local_date_string(section.reservations_end_date),
+            applied_reservations_per_week=str(section.applied_reservations_per_week),
+            reservation_min_duration=local_timedelta_string(section.reservation_min_duration),
+            reservation_max_duration=local_timedelta_string(section.reservation_max_duration),
+        )
+
+        # Applicant
+        if section.application.organisation is not None:
+            row.applicant = section.application.organisation.name
+        elif (contact_person := section.application.contact_person) is not None:
+            row.applicant = f"{contact_person.first_name} {contact_person.last_name}"
+
+        # Reservation Unit Options
+        for option in section.reservation_unit_options.all():
+            option_row = ApplicationSectionExportRow(**asdict(row))
+            option_row.reservation_unit_name = option.reservation_unit.name
+            option_row.unit_name = option.reservation_unit.unit.name
+
+            # One row per Allocated Time Slot
+            if len(option.allocated_time_slots.all()) == 0:
+                # An option without allocated time slots must still be exported
+                yield option_row
```

Comment:
An option row without any allocations should not be exported by itself, but if there are no allocations at all across all the options rows of an application section, then a single "section_row" (i.e., spanning all its options) needs to be exported so that the section is represented once in the export.

Reply:
Added a fix for this 👍